src/common/tracer.cc

```diff
@@ -35,15 +35,14 @@ void Tracer::shutdown() {
   }
 }
 
-jspan Tracer::start_trace(opentelemetry::nostd::string_view trace_name) {
-  if (is_enabled()) {
+jspan Tracer::start_trace(opentelemetry::nostd::string_view trace_name, bool trace_is_enabled) {
+  if (trace_is_enabled && this->is_enabled()) {
     return tracer->StartSpan(trace_name);
   }
   return noop_tracer->StartSpan(trace_name);
 }
-
 jspan Tracer::add_span(opentelemetry::nostd::string_view span_name, const jspan& parent_span) {
-  if (is_enabled() && parent_span) {
+  if (is_enabled() && parent_span->IsRecording()) {
```

Comment:
why did you make this change here?
can't the parent span be null?
does `IsRecording()` return `false` in case of noop span?

Reply:
parent span can not be null. it is initialized earlier in every scenario (`start_trace` is not a conditional call).  

the first idea was to check if the parent span is a noop span, but before this PR it was impossible to create a noop span while the tracer is enabled. now it's a possibility. so `isRecording()` return false if the parent span is a noop, while the tracer is enabled.
so it won't send spans with an invalid parent.